test_programs/plonky2_prove_failure/signed_casts/src/main.nr

```diff
@@ -0,0 +1,10 @@
+fn main(x: [i8; 2], y: [i16; 2], z: [i32; 2]) {
```

Comment:
I'm not sure if Noir support 16-bit integer types? Rust does support them, but Noir doesn't, as far as I know. I think it only supports 8-bit, 32-bit and 64-bit. And 64-bit don't work with the PLONKY2 backend, so the only remaining conversion we can test is i8 -> i32.

Reply:
In the [noir documentation](https://noir-lang.org/docs/dev/noir/concepts/data_types/integers) it is stated:

> An integer type is a range constrained field type. The Noir frontend supports both unsigned and signed integer types. The allowed sizes are 1, 8, 16, 32 and 64 bits.
> [...]
> A signed integer type is specified first with the letter i (which stands for integer) followed by its bit size (e.g. 8)

I couldn't find anywhere it being said that 16-bit signed integers aren't supported.

Reply:
I tested it, and it seems to be supported. I was mislead by the error message "Allowed bit sizes for integers are 1, 8, 32, 64" that the compiler produces, when you try to use some non-standard bit size, such as u63.